### PR TITLE
Add ol.proj.setProj4 function

### PR DESCRIPTION
**Pull request by bartvde**
closes #4230 and #3990

### CODE REVIEW

File: changelog/upgrade-notes.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
 ## Upgrade notes
 
+### v3.13.0
+
+#### `proj4js` integration
+
+Before this release, OpenLayers depended on the global proj4 namespace. When using for instance browserify you might not want to depend on the global `proj4` namespace. You can use the `ol.proj.setProj4` function to set the proj4 function object. For example in a browserify ES6 environment:
```
Comment (ahocevar) [inline changelog/upgrade-notes.md]: Maybe move (or copy) some of this (especially the code snippet) to the `ol.proj.setProj4` API docs.

Comment (bartvde) [inline changelog/upgrade-notes.md]: Good point, will do.

File: changelog/upgrade-notes.md [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+
+```js
+import ol from 'openlayers';
+import proj4 from 'proj4';
+proj4.defs('EPSG:28992', '+proj=sterea +lat_0=52.15616055555555 +lon_0=5.38763888888889 +k=0.9999079 +x_0=155000 +y_0=463000 +ellps=bessel +units=m +towgs84=565.2369,50.0087,465.658,-0.406857330322398,0.350732676542563,-1.8703473836068,4.0812 +no_defs');
```
Comment (ahocevar) [inline changelog/upgrade-notes.md]: I think this line is not relevant.

**Comment by ahocevar** [comment]
Thanks @bartvde, looks good. I only added two doc comments. Please merge when addressed and when Travis agrees.

File: src/ol/proj/proj.js [hunk 3]
```diff
@@ -75,7 +75,8 @@ ol.proj.METERS_PER_UNIT[ol.proj.Units.USFEET] = 1200 / 3937;
  *     http://www.opengis.net/gml/srs/epsg.xml#3857
  *
  * If you use proj4js, aliases can be added using `proj4.defs()`; see
- * [documentation](https://github.com/proj4js/proj4js).
+ * [documentation](https://github.com/proj4js/proj4js). To set an alternative
+ * namespace for proj4, use `ol.proj.setProj4`.
```
Comment (ahocevar) [inline src/ol/proj/proj.js]: You could use `{@link ol.proj.setProj4}`

File: src/ol/proj/proj.js [hunk 4]
```diff
@@ -406,6 +409,34 @@ ol.proj.transforms_ = {};
...
+ * browserify ES6 environment you could use:
+ *
+ *     import ol from 'openlayers';
+ *     import proj4 from 'proj4';
+ *     ol.proj.setProj4(proj4);
+ *
+ * @param {proj4} proj4 Proj4.
+ * @api
+ */
+ol.proj.setProj4 = function(proj4) {
```
Comment (ahocevar) [inline src/ol/proj/proj.js]: Would gjslint or the compiler complain if this is flipped?
```js
if (ol.ENABLE_PROJ4JS) {
  ol.proj.setProj4 = function(proj4) {
    // ...
  }
}
```

Comment (bartvde) [inline src/ol/proj/proj.js]: No that works fine, so will make the change.

**Comment by ahocevar** [comment]
Ok thanks, then please merge when everything is green.

**Comment by juristr** [comment]
:+1: Awesome, thx @fredj for this fix!

**Pull request merged**